[TZone] WebCore/inspector Convert FastMalloc to TZone

msaboff opened the pull request:
#### 9a4e00abd622d7419587b20e1d4d5d98bae69f75
<pre>
[TZone] WebCore/inspector Convert FastMalloc to TZone
<a href="https://bugs.webkit.org/show_bug.cgi?id=278430">https://bugs.webkit.org/show_bug.cgi?id=278430</a>
<a href="https://rdar.apple.com/134377811">rdar://134377811</a>

Reviewed by Devin Rousso.

Convert WTF_MAKE_FAST_ALLOCATED to WTF_MAKE_TZONE_ALLOCATED in
preparation for enabling TZone (not yet enabled).

* Source/WebCore/inspector/CommandLineAPIHost.cpp:
* Source/WebCore/inspector/CommandLineAPIHost.h:
* Source/WebCore/inspector/DOMEditor.cpp:
* Source/WebCore/inspector/DOMEditor.h:
* Source/WebCore/inspector/InspectorController.cpp:
* Source/WebCore/inspector/InspectorController.h:
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/inspector/InspectorHistory.cpp:
* Source/WebCore/inspector/InspectorHistory.h:
* Source/WebCore/inspector/InspectorOverlay.cpp:
* Source/WebCore/inspector/InspectorOverlay.h:
* Source/WebCore/inspector/InspectorOverlayLabel.cpp:
* Source/WebCore/inspector/InspectorOverlayLabel.h:
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
* Source/WebCore/inspector/InstrumentingAgents.cpp:
* Source/WebCore/inspector/InstrumentingAgents.h:
* Source/WebCore/inspector/NetworkResourcesData.cpp:
* Source/WebCore/inspector/NetworkResourcesData.h:
* Source/WebCore/inspector/PageDebugger.cpp:
* Source/WebCore/inspector/PageDebugger.h:
* Source/WebCore/inspector/UserGestureEmulationScope.cpp:
* Source/WebCore/inspector/UserGestureEmulationScope.h:
* Source/WebCore/inspector/WebInjectedScriptManager.cpp:
* Source/WebCore/inspector/WebInjectedScriptManager.h:
* Source/WebCore/inspector/WorkerDebugger.cpp:
* Source/WebCore/inspector/WorkerDebugger.h:
* Source/WebCore/inspector/WorkerInspectorController.cpp:
* Source/WebCore/inspector/WorkerInspectorController.h:
* Source/WebCore/inspector/WorkerToPageFrontendChannel.h:
* Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
* Source/WebCore/inspector/agents/InspectorAnimationAgent.h:
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.cpp:
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.h:
* Source/WebCore/inspector/agents/InspectorCPUProfilerAgent.cpp:
* Source/WebCore/inspector/agents/InspectorCPUProfilerAgent.h:
* Source/WebCore/inspector/agents/InspectorCSSAgent.cpp:
* Source/WebCore/inspector/agents/InspectorCSSAgent.h:
* Source/WebCore/inspector/agents/InspectorCanvasAgent.cpp:
* Source/WebCore/inspector/agents/InspectorCanvasAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMStorageAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDOMStorageAgent.h:
* Source/WebCore/inspector/agents/InspectorDatabaseAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDatabaseAgent.h:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
(WebCore::Inspector::ExecutableWithDatabase::ExecutableWithDatabase): Deleted.
(WebCore::Inspector::ExecutableWithDatabase::context const): Deleted.
(WebCore::Inspector::ExecutableWithDatabase::start): Deleted.
(WebCore::Inspector::keyPathFromIDBKeyPath): Deleted.
(WebCore::Inspector::transactionForDatabase): Deleted.
(WebCore::Inspector::objectStoreForTransaction): Deleted.
(WebCore::Inspector::indexForObjectStore): Deleted.
(WebCore::Inspector::idbKeyFromInspectorObject): Deleted.
(WebCore::Inspector::idbKeyRangeFromKeyRange): Deleted.
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.h:
* Source/WebCore/inspector/agents/InspectorLayerTreeAgent.cpp:
* Source/WebCore/inspector/agents/InspectorLayerTreeAgent.h:
* Source/WebCore/inspector/agents/InspectorMemoryAgent.cpp:
* Source/WebCore/inspector/agents/InspectorMemoryAgent.h:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(): Deleted.
(WebCore::Inspector::buildWebSocketMessage): Deleted.
* Source/WebCore/inspector/agents/InspectorNetworkAgent.h:
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
* Source/WebCore/inspector/agents/InspectorPageAgent.h:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.h:
* Source/WebCore/inspector/agents/InspectorWorkerAgent.cpp:
* Source/WebCore/inspector/agents/InspectorWorkerAgent.h:
* Source/WebCore/inspector/agents/WebDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/WebDebuggerAgent.h:
* Source/WebCore/inspector/agents/WebHeapAgent.cpp:
* Source/WebCore/inspector/agents/WebHeapAgent.h:
* Source/WebCore/inspector/agents/page/PageAuditAgent.cpp:
* Source/WebCore/inspector/agents/page/PageAuditAgent.h:
* Source/WebCore/inspector/agents/page/PageCanvasAgent.cpp:
* Source/WebCore/inspector/agents/page/PageCanvasAgent.h:
* Source/WebCore/inspector/agents/page/PageConsoleAgent.cpp:
* Source/WebCore/inspector/agents/page/PageConsoleAgent.h:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.h:
* Source/WebCore/inspector/agents/page/PageHeapAgent.cpp:
* Source/WebCore/inspector/agents/page/PageHeapAgent.h:
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:
* Source/WebCore/inspector/agents/page/PageNetworkAgent.h:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.h:
* Source/WebCore/inspector/agents/worker/ServiceWorkerAgent.cpp:
* Source/WebCore/inspector/agents/worker/ServiceWorkerAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerAuditAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerAuditAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerCanvasAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerCanvasAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerConsoleAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerConsoleAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerDebuggerAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerRuntimeAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerRuntimeAgent.h:

Canonical link: <a href="https://commits.webkit.org/282574@main">https://commits.webkit.org/282574@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4441c87ae620c287bf6f5768028116dba5bfa996

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63432 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42788 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16029 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67453 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14040 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65552 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50476 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14320 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51079 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9703 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66501 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39719 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54927 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31765 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36396 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12284 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12912 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57927 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12608 "Found 1 new test failure: imported/w3c/web-platform-tests/websockets/basic-auth.any.serviceworker.html?wss (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69149 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7379 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12196 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58386 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7411 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55002 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58626 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/14064 "The change is no longer eligible for processing.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6150 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9610 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38609 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39688 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40800 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39431 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
```diff
@@ -100,6 +101,10 @@ namespace WebCore {
 using namespace Inspector;
 
+WTF_MAKE_TZONE_ALLOCATED_IMPL(InspectorNetworkAgent);
+WTF_MAKE_TZONE_ALLOCATED_IMPL_NESTED(InspectorNetworkAgentPendingInterceptRequest, InspectorNetworkAgent::AgentPendingInterceptRequest);
```
typo with `PendingInterceptRequest`

dcrousso reviewed (approved):
should these be added to the objects that don't currently have them (e.g. `WorkerDOMDebuggerAgent`)?

msaboff committed:
[TZone] WebCore/inspector Convert FastMalloc to TZone
https://bugs.webkit.org/show_bug.cgi?id=278430
rdar://134377811

Reviewed by Devin Rousso.

Convert WTF_MAKE_FAST_ALLOCATED to WTF_MAKE_TZONE_ALLOCATED in
preparation for enabling TZone (not yet enabled).

* Source/WebCore/inspector/CommandLineAPIHost.cpp:
* Source/WebCore/inspector/CommandLineAPIHost.h:
* Source/WebCore/inspector/DOMEditor.cpp:
* Source/WebCore/inspector/DOMEditor.h:
* Source/WebCore/inspector/InspectorController.cpp:
* Source/WebCore/inspector/InspectorController.h:
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/inspector/InspectorHistory.cpp:
* Source/WebCore/inspector/InspectorHistory.h:
* Source/WebCore/inspector/InspectorOverlay.cpp:
* Source/WebCore/inspector/InspectorOverlay.h:
* Source/WebCore/inspector/InspectorOverlayLabel.cpp:
* Source/WebCore/inspector/InspectorOverlayLabel.h:
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
* Source/WebCore/inspector/InstrumentingAgents.cpp:
* Source/WebCore/inspector/InstrumentingAgents.h:
* Source/WebCore/inspector/NetworkResourcesData.cpp:
* Source/WebCore/inspector/NetworkResourcesData.h:
* Source/WebCore/inspector/PageDebugger.cpp:
* Source/WebCore/inspector/PageDebugger.h:
* Source/WebCore/inspector/UserGestureEmulationScope.cpp:
* Source/WebCore/inspector/UserGestureEmulationScope.h:
* Source/WebCore/inspector/WebInjectedScriptManager.cpp:
* Source/WebCore/inspector/WebInjectedScriptManager.h:
* Source/WebCore/inspector/WorkerDebugger.cpp:
* Source/WebCore/inspector/WorkerDebugger.h:
* Source/WebCore/inspector/WorkerInspectorController.cpp:
* Source/WebCore/inspector/WorkerInspectorController.h:
* Source/WebCore/inspector/WorkerToPageFrontendChannel.h:
* Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
* Source/WebCore/inspector/agents/InspectorAnimationAgent.h:
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.cpp:
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.h:
* Source/WebCore/inspector/agents/InspectorCPUProfilerAgent.cpp:
* Source/WebCore/inspector/agents/InspectorCPUProfilerAgent.h:
* Source/WebCore/inspector/agents/InspectorCSSAgent.cpp:
* Source/WebCore/inspector/agents/InspectorCSSAgent.h:
* Source/WebCore/inspector/agents/InspectorCanvasAgent.cpp:
* Source/WebCore/inspector/agents/InspectorCanvasAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMStorageAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDOMStorageAgent.h:
* Source/WebCore/inspector/agents/InspectorDatabaseAgent.cpp:
* Source/WebCore/inspector/agents/InspectorDatabaseAgent.h:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
(WebCore::Inspector::ExecutableWithDatabase::ExecutableWithDatabase): Deleted.
(WebCore::Inspector::ExecutableWithDatabase::context const): Deleted.
(WebCore::Inspector::ExecutableWithDatabase::start): Deleted.
(WebCore::Inspector::keyPathFromIDBKeyPath): Deleted.
(WebCore::Inspector::transactionForDatabase): Deleted.
(WebCore::Inspector::objectStoreForTransaction): Deleted.
(WebCore::Inspector::indexForObjectStore): Deleted.
(WebCore::Inspector::idbKeyFromInspectorObject): Deleted.
(WebCore::Inspector::idbKeyRangeFromKeyRange): Deleted.
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.h:
* Source/WebCore/inspector/agents/InspectorLayerTreeAgent.cpp:
* Source/WebCore/inspector/agents/InspectorLayerTreeAgent.h:
* Source/WebCore/inspector/agents/InspectorMemoryAgent.cpp:
* Source/WebCore/inspector/agents/InspectorMemoryAgent.h:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(): Deleted.
(WebCore::Inspector::buildWebSocketMessage): Deleted.
* Source/WebCore/inspector/agents/InspectorNetworkAgent.h:
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
* Source/WebCore/inspector/agents/InspectorPageAgent.h:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.h:
* Source/WebCore/inspector/agents/InspectorWorkerAgent.cpp:
* Source/WebCore/inspector/agents/InspectorWorkerAgent.h:
* Source/WebCore/inspector/agents/WebDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/WebDebuggerAgent.h:
* Source/WebCore/inspector/agents/WebHeapAgent.cpp:
* Source/WebCore/inspector/agents/WebHeapAgent.h:
* Source/WebCore/inspector/agents/page/PageAuditAgent.cpp:
* Source/WebCore/inspector/agents/page/PageAuditAgent.h:
* Source/WebCore/inspector/agents/page/PageCanvasAgent.cpp:
* Source/WebCore/inspector/agents/page/PageCanvasAgent.h:
* Source/WebCore/inspector/agents/page/PageConsoleAgent.cpp:
* Source/WebCore/inspector/agents/page/PageConsoleAgent.h:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.h:
* Source/WebCore/inspector/agents/page/PageHeapAgent.cpp:
* Source/WebCore/inspector/agents/page/PageHeapAgent.h:
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:
* Source/WebCore/inspector/agents/page/PageNetworkAgent.h:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.h:
* Source/WebCore/inspector/agents/worker/ServiceWorkerAgent.cpp:
* Source/WebCore/inspector/agents/worker/ServiceWorkerAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerAuditAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerAuditAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerCanvasAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerCanvasAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerConsoleAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerConsoleAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerDebuggerAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerRuntimeAgent.cpp:
* Source/WebCore/inspector/agents/worker/WorkerRuntimeAgent.h:

Canonical link: https://commits.webkit.org/282574@main
